### PR TITLE
Add example for no_std support

### PR DESCRIPTION
**Pull request by mre**
This uses [cortex-m-quickstart](https://github.com/rust-embedded/cortex-m-quickstart) as suggested by @thejpster in https://github.com/mre/mos6502/pull/10#issuecomment-435651905.

### CODE REVIEW

**Comment by mre** [comment]
The @thejpster, does this make any sense to you?

**Comment by thejpster** [comment]
LGTM, from an Embedded Rust point of view.

What does your example 6502 machine code do?

File: no-std-example/memory.x [hunk 1]
```diff
@@ -0,0 +1,34 @@
+MEMORY
+{
+  /* NOTE 1 K = 1 KiBi = 1024 bytes */
+  /* TODO Adjust these memory regions to match your device memory layout */
+  /* These values correspond to the LM3S6965, one of the few devices QEMU can emulate */
+  FLASH : ORIGIN = 0x00000000, LENGTH = 256K
+  RAM : ORIGIN = 0x20000000, LENGTH = 64K
```
Comment (mre) [inline no-std-example/memory.x:7]: What should we set this to? 🤔

**Comment by mre** [comment]
Can't believe I'm coming back to this PR after such a long time. I got a bit carried away by #30 and decided to brush up that old project a bit.

Sorry @thejpster, I don't have much experience with emulation, but I'd love to get this merged... eventually. 😆 
So my apologies but do you know what I could set for the RAM here?
https://github.com/mre/mos6502/pull/16/files#r566457164
I've set it to 128K now and it works. Probably that should be fine?

Totally understandable if you don't have time to answer. Just wanted to give it a shot.

> Edit 2: The Cpu object is also on the stack - objects of this size would usually be statically allocated.

Would this work?

```
use once_cell::unsync::Lazy;

#[entry]
fn main() -> ! {
    let mut cpu = Lazy::new(|| cpu::CPU::new());
    // ...
}
```
[Reference](https://github.com/mre/mos6502/pull/16/files#diff-e1a5636818b95b1ad3a2ff1519ebef8bd92634035a470d83e22d7dcf3af5a2a0R17)

If not, do you have an example how to statically allocate the CPU?

**Pull request closed without merging**